File: IAIso-v5.0/core/iaiso-php/src/Identity/Jwk.php
<?php

declare(strict_types=1);

namespace IAIso\Identity;

/** A single key from a JWKS document. */
final class Jwk
{
    public function __construct(
        public readonly string $kty,
        public readonly ?string $kid = null,
        public readonly ?string $alg = null,
        public readonly ?string $use = null,
        public readonly ?string $n = null,
        public readonly ?string $e = null,
    ) {}
}

File: IAIso-v5.0/core/iaiso-ruby/lib/iaiso/identity/jwk.rb
# frozen_string_literal: true

module IAIso
  module Identity
    # A single JWK key entry.
    Jwk = Data.define(:kty, :kid, :alg, :use, :n, :e) do
      def self.from_hash(h)
        new(
          kty: h["kty"].to_s,
          kid: h["kid"],
          alg: h["alg"],
          use: h["use"],
          n: h["n"],
          e: h["e"],
        )
      end
    end

    # JWKS document.
    Jwks = Data.define(:keys)
  end
end
